File: app/src/test/java/io/github/freewebmovement/igniter/APITest.java
package io.github.freewebmovement.igniter;

import androidx.test.runner.AndroidJUnit4;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;
import org.junit.Test;
import org.junit.runner.RunWith;

import io.github.freewebmovement.igniter.connection.API;
import io.github.freewebmovement.igniter.models.Server;

@RunWith(AndroidJUnit4.class)

public class APITest {

    @Test
    public void shouldGetServer() throws JSONException {
        API api = new API();
        Server[] servers = api.server();
        assert (servers.length >= 1);

    }
    @Test
    public void shouldGetQuota() throws JSONException {
        String username = "sammy";
        String password = "1234";
        API api = new API();
        String quotaStr = api.quota(username, password);
        JSONObject quotaJSON = new JSONObject(quotaStr);
        assert (quotaJSON.getString("username").equals(username));
        assert (quotaJSON.getInt("quota") == 0);
        assert (quotaJSON.getInt("upload") == 0);
        assert (quotaJSON.getInt("download") == 0);
    }
}

File: app/src/test/java/io/github/freewebmovement/igniter/APITest.kt
package io.github.freewebmovement.igniter

import androidx.test.runner.AndroidJUnit4
import io.github.freewebmovement.igniter.connection.API
import org.json.JSONObject
import org.junit.Test
import org.junit.runner.RunWith

@RunWith(AndroidJUnit4::class)
class APITest {

    @Test
    fun shouldGetServer() {
        val api = API()
        val servers = api.server()
        assert(servers.size >= 1)
    }

    @Test
    fun shouldGetQuota() {
        val username = "sammy"
        val password = "1234"
        val api = API()
        val quotaStr = api.quota(username, password)
        val quotaJSON = JSONObject(quotaStr)
        assert(quotaJSON.getString("username") == username)
        assert(quotaJSON.getInt("quota") == 0)
        assert(quotaJSON.getInt("upload") == 0)
        assert(quotaJSON.getInt("download") == 0)
    }
}
